src/components/AssetIcon.tsx

```diff
@@ -80,19 +95,10 @@ export const AssetIcon = ({ assetId, showNetworkIcon, src, ...rest }: AssetIconP
       )
     }
 
-    if (assetId === nativeAssetId && asset?.networkIcon && showNetworkIcon)
-      return (
-        <Avatar
-          src={asset.networkIcon}
-          bg={assetIconBg}
-          icon={<FoxIcon boxSize='16px' color={assetIconColor} />}
-          {...rest}
-        />
-      )
-
     return (
       <AssetWithNetwork
         assetId={assetId}
+        src={src}
```

Comment:
added so we can pass through the nft token src in lieu of being able to dynamically add ntf assets to the assetSlice. I was seeing re-render problems when attempting to do this rendering tx history, but may have been doing it incorrectly..

Reply:
Note this will now consistently use the `src` in place of the asset icon, so long as `src` is defined

Reply:
Still believe it would make more sense to store as an actual asset - for e.g sends, or being able to see an "asset page" for the current NFT if that's part of the upcoming spec (i.e with collection etc), then we'll need an actual asset

Reply:
understood `src` will be used in this context for now and agree we want to store the assets ultimately. down to sync on this to see if we can get that sorted here or fast follow